lib/efi.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * EFI-related functions to set up and run test cases in EFI
+ *
+ * Copyright (c) 2021, SUSE, Varad Gautam <varad.gautam@suse.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.0-or-later
+ */
+#include <linux/efi.h>
+
+unsigned long __efiapi efi_main(efi_handle_t handle, efi_system_table_t *sys_tab);
+efi_system_table_t *efi_system_table = NULL;
+
+static void efi_free_pool(void *ptr)
+{
+	efi_bs_call(free_pool, ptr);
+}
+
+static efi_status_t efi_get_memory_map(struct efi_boot_memmap *map)
+{
+	efi_memory_desc_t *m = NULL;
+	efi_status_t status;
+	unsigned long key = 0, map_size = 0, desc_size = 0;
+
+	status = efi_bs_call(get_memory_map, &map_size,
+			     NULL, &key, &desc_size, NULL);
+	if (status != EFI_BUFFER_TOO_SMALL || map_size == 0)
+		goto out;
+
+	/* Pad map_size with additional descriptors so we don't need to retry */
+	map_size += 4 * desc_size;
+	*map->buff_size = map_size;
+	status = efi_bs_call(allocate_pool, EFI_LOADER_DATA,
```

Comment:
This fine. But I guess the problem with the initial comment wasn't necessarily that it spanned two lines. The issue is that a multi-line Linux style comment should have the opening `/*` and closing `*/` on their own lines.

Linux coding style:
https://www.kernel.org/doc/html/latest/process/coding-style.html#commenting

Random source code example:
https://elixir.bootlin.com/linux/latest/source/arch/x86/kvm/svm/svm.c#L83

upstreaming-v2:
> +	/* Pad map_size with additional descriptors so we don't need to
> +	 * retry. */

Linux style:
	/*
	 * Pad map_size with additional descriptors so we don't need to
	 * retry.
	 */

Reply:
Ah yes, I discovered it can squeeze into one line (120 line size, cannot fit 80 line size). kvm-unit-tests style supports [120 line size](https://github.com/marc-orr/KVM-Unit-Tests-dev-fork/blob/master/README.md#style).

But since this file is already assuming 80 line size, I will change it back to a Linux-style multiline comment.

Reply:
Either way is fine. I just wanted to make sure you fully understood the options. One line is fine with me.

Reply:
I changed it back to multi-line comment in Linux style.